Water-Physics/Assets/Code/Scripts/Environment/Waves.cs

```diff
@@ -0,0 +1,180 @@
+        var dist = (max - Vector3.Distance(p1, localPos))
+                 + (max - Vector3.Distance(p2, localPos))
+                 + (max - Vector3.Distance(p3, localPos))
+                 + (max - Vector3.Distance(p4, localPos) + Mathf.Epsilon);
+        //weighted sum
+        var height = _mesh.vertices[Index(Mathf.CeilToInt(p1.x), Mathf.CeilToInt(p1.z))].y * (max - Vector3.Distance(p1, localPos))
+                     + _mesh.vertices[Index(Mathf.CeilToInt(p2.x), Mathf.CeilToInt(p2.z))].y * (max - Vector3.Distance(p2, localPos))
+                     + _mesh.vertices[Index(Mathf.CeilToInt(p3.x), Mathf.CeilToInt(p3.z))].y * (max - Vector3.Distance(p3, localPos))
+                     + _mesh.vertices[Index(Mathf.CeilToInt(p4.x), Mathf.CeilToInt(p4.z))].y * (max - Vector3.Distance(p4, localPos));
+
+        //scale
+        return height * transform.lossyScale.y / dist;
+
+    }
+
+        // Start is called before the first frame update
+        private void Start()
+        {
+            _mesh = new Mesh();
+            _mesh.name = gameObject.name;
+
+            _mesh.vertices = GenerateVertices();
+            _mesh.triangles = GenerateTriangles();
+            _mesh.uv = GenerateUVs();
+
+            _mesh.RecalculateBounds();
+            _mesh.RecalculateNormals();
+
+            _meshFilter = gameObject.AddComponent<MeshFilter>();
+            _meshFilter.mesh = _mesh;
+        }
+
+        private Vector2[] GenerateUVs()
+        {
+            var uvs = new Vector2[_mesh.vertices.Length];
+
+            //Always set one UV over n tiles then flip the UV and set it again
+            for (int x = 0; x <= dimension; x++)
+            {
+                for (int z = 0; z <= dimension; z++)
+                {
+                    var vector = new Vector2((x / uvScale) % 2, (z / uvScale) % 2);
+                    uvs[Index(x, z)] = new Vector2(vector.x <= 1 ? vector.x : 2 - vector.x,
+                        vector.y <= 1 ? vector.y : 2 - vector.y);
+                }
+            }
+
+            return uvs;
+        }
+
+        private int[] GenerateTriangles()
+        {
+            var triangles = new int[_mesh.vertices.Length * 6];
+
+            //Two triangles per tile
+            for (int x = 0; x < dimension; x++)
+            {
+                for (int z = 0; z < dimension; z++)
+                {
+                    triangles[Index(x, z) * 6 + 0] = Index(x, z);
+                    triangles[Index(x, z) * 6 + 1] = Index(x + 1, z + 1);
+                    triangles[Index(x, z) * 6 + 2] = Index(x + 1, z);
+                    triangles[Index(x, z) * 6 + 3] = Index(x, z);
+                    triangles[Index(x, z) * 6 + 4] = Index(x, z + 1);
+                    triangles[Index(x, z) * 6 + 5] = Index(x + 1, z + 1);
+                }
+            }
+
+            return triangles;
+        }
```

Comment:
Ik zie hier heel veel getallen uit het niets komen. Staan deze vast voor de berekeningen die je uitvoert, of kunnen deze nog veranderen als bijvoorbeeld de mesh groter maakt? Anders moet je misschien op een andere manier aan deze getallen komen.

Reply:
Nee, deze nummers zijn voor elk deel hetzelfde.